lib/xapi-stdext-zerocheck/zerocheck_stub.c

```diff
@@ -108,7 +108,7 @@ value find_a_zero(value string, value offset, value remaining)
 value is_all_zeros(value string, value length)
 {
 	CAMLparam2(string, length);
-	char *s = String_val(string);
+	char *s = Bytes_val(string);
```

Comment:
this could remain a string if all the other pointers are changed to `const`. Since this function is not meant to modify its arguments it might be better to do it that way.